project/src/components/main-page/main-page.tsx

```diff
@@ -2,13 +2,24 @@ import Header from '../header/header';
 import {Link} from 'react-router-dom';
 import {OfferType} from '../../types/offer';
 import PlaceCardList from '../place-card-list/place-card-list';
+import { CITY } from '../../mocks/city';
+import Map from '../map/map';
+import {useState} from 'react';
 
 type MainPageProps = {
   rentOffer: number;
   offers: OfferType[];
 }
 
 function MainPage({rentOffer, offers}:MainPageProps): JSX.Element {
+  const [selectedPoint, setSelectedPoint] = useState<OfferType | null>(null);
+
+  const onPlaceCardHover = (id: number) => {
+    const selectedId = offers.find((offer) => offer.id === id) ?? null;
+
+    setSelectedPoint(selectedId);
```

Comment:
```suggestion
  const onPlaceCardHover = (offer: OfferType) => {
    setSelectedPoint(offer);
```
а почему бы тебе сюда просто весь оффер не передавать?